src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt

```diff
@@ -81,10 +101,100 @@ object MovementListener : Listener {
 
     private fun teleportBetweenSections(player: Player, to: Location, oldSection: Section, newSection: Section) {
         val newLoc = to.getCorrespondingLocation(oldSection, newSection) ?: return
-        val fallDistance = player.fallDistance
-        val oldVelocity = player.velocity
-        player.teleport(newLoc)
-        player.fallDistance = fallDistance
-        player.velocity = oldVelocity
+
+        val oldLeashedEntities = player.getLeashedEntities()
+        oldLeashedEntities?.forEach {
```

Comment:
Is the point of this to prevent entities from despawning? If so, an entity being named doesn't guarantee no despawn happening. You might be able to toggle whether the entity despawns with `removeWhenFarAway` iirc, but from experimentation I'm pretty sure entities only despawn when the chunk unloads and chunks don't unload immediately (that might have been an option we set ourselves in paper config).

Either way, if the code works without this, leave it be. If it doesn't we need to be a bit more careful here.

Reply:
I assume this is about me assigning custom names to entities. I'll check if the functionality is fine without it. But if the unloading of chunks is a config setting, wouldn't it be nice if this works regardless of such a setting?

Reply:
The functionality seems fine. I'll remove the custom name stuff then.